src/Raven.Client/Documents/Session/Operations/QueryOperation.cs

```diff
@@ -273,6 +273,21 @@ public void EnsureIsAcceptableAndSaveResult(QueryResult result)
 
             EnsureIsAcceptable(result, _indexQuery.WaitForNonStaleResults, _sp, _session);
 
+            SaveQueryResult(result);
+        }
+
+        public void EnsureIsAcceptableAndSaveResult(QueryResult result, TimeSpan responseElapsed)
```

Comment:
rename parameter to `duration`

Reply:
Maybe change to `internal` and make `duration` nullable? Then from the other `EnsureIsAcceptableAndSaveResult` call this method, but pass `null` for duration

Reply:
done